### ISSUE TITLE
Memory leak when run benchmark test of get with small object and high concurrency

### ISSUE THREAD

**Issue by huangp0600** [minio/warp#113]
I use warp to do benchmark test of get. But I notice some strange things.

**large object(size=10M, duration=10m, concurrency=64)**
When I do the test with large object, warp runs good. 
![image](https://user-images.githubusercontent.com/8937150/85358317-f40bb400-b545-11ea-86f7-c53ad5596205.png)

**small object(size=64K, duration=10m, concurrency=64)**
Meanwhile if I do it with small object, warp would swallow my computer's memory gradually. Finally it occupies all the memory. Then some kind of exception was thrown continuously.
![image](https://user-images.githubusercontent.com/8937150/85358231-cc1c5080-b545-11ea-8264-c44cf2ebcfc5.png)

the exception is,
```
goroutine 3847466 [select, 12 minutes]:
github.com/minio/minio-go/v6.Client.getObjectWithContext.func1(0xc5d3af4d80, 0xc5d3af4de0, 0xc5d3af4e40, 0xc5b8c44c50, 0xc5d3b467a0, 0xc5d2e350e0, 0xc04260, 0xc0000261b0, 0x7ffe338ac7bf, 0x22, ...)
    /home/phuang1/go/pkg/mod/github.com/minio/minio-go/v6@v6.0.55/api-get-object.go:75 +0x18c
created by github.com/minio/minio-go/v6.Client.getObjectWithContext
    /home/phuang1/go/pkg/mod/github.com/minio/minio-go/v6@v6.0.55/api-get-object.go:66 +0x346
```

the cmd is as follows,
./warp get --host=xxx --access-key=xxx --secret-key=xxx --obj.size=64KiB --concurrent=64
    --duration=10m

**Comment by klauspost** [comment]
Most likely a problem with `mino-go`, but using this to identify the issue.

**Comment by klauspost** [comment]
Nope problem in warp. Fixed in #115 - thanks for reporting!

**Comment by huangp0600** [comment]
Thanks for quick response! The fix seems good. I do the test again with this latest fixed version. 
the para is: `--size=64k  --concurrency=64  --duration=30m`
The memory usage ascends with a more flat curve.
![image](https://user-images.githubusercontent.com/8937150/85643507-f1849800-b6c6-11ea-84a9-2e3a730e0fc4.png)

Not clear if it will reach the max memory if I prolong the duration time.
however I also haven't found other problems except the `Close` which would increase the memory usage potentially at present.
